### PR TITLE
[BugFix] fix NPE if hostname can't be resolved

### PR DESCRIPTION
**Pull request by kevincai**
* don't panic if hostname can't be resolved to ip address

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

## What type of PR is this：
- [X] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java [hunk 1]
```diff
@@ -147,7 +147,11 @@ private static boolean isJoin(List<InetSocketAddress> allFeHosts, Frontend fe) {
             if (InetAddressValidator.getInstance().isValidInet4Address(fe.getHost())) {
                 realHost = addr.getAddress().getHostAddress();
             } else {
-                realHost = addr.getAddress().getHostName();
+                try {
+                    realHost = addr.getAddress().getHostName();
+                } catch (Exception e) {
+                    LOG.warn("Fail to get address hostname of fe: " + addr.toString() + " msg:" + e.toString());
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java:153]:
```suggestion
                    LOG.warn("Fail to get address hostname of fe: {} msg: {}", addr, e);
```
the last e is used to print the error stack.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java (hunk 1 above)
Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java]: done

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by imay — APPROVED** [review]
LGTM

**Comment by kevincai** [comment]
need backport to branch-2.4 due to FQDN feature released since v2.4

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|4|4|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**